packages/client/src/pages/main/modules/profile/Profile.tsx

```diff
@@ -8,136 +8,142 @@ import { FormField } from './components/FormField'
 import { validationSchema } from './validation'
 import { DEFAULT_USER_DATA } from './constants'
 import { useGetUserInfoQuery } from '../../../../store/api/yadnex/auth/authApi'
-import { UserFullInfo } from '../../../../types/auth'
-import { useNavigate } from 'react-router-dom'
-import { GAME_ROUTE } from '../../../../const/route'
+import { UserProfileInfo } from '../../../../types/auth'
+import PasswordForm from './modules/PasswordForm'
+import { RESOURCES_HOST } from '../../../../const/host'
+import { useChangeProfileMutation } from '../../../../store/api/yadnex/profile/profileApi'
+import { toast } from 'react-toastify'
+import defaultAvatar from '../../../../assets/img/default-avatar.png';
 
 const Profile: React.FC = () => {
-  const [user, setUser] = useState<UserFullInfo>(DEFAULT_USER_DATA)
+  const [user, setUser] = useState<UserProfileInfo>(DEFAULT_USER_DATA)
   const [isAvatarShown, setIsAvatarShown] = useState<boolean>(false);
   const {data} = useGetUserInfoQuery()
-  const navigate = useNavigate()
 
   useEffect(() => {
     if (data) {
       setUser(data)
     }
   }, [user, data])
+  
+  const [changeProfile] = useChangeProfileMutation();
+  
 
-  const handleSubmit = (data: UserFullInfo) => {
-    console.log('submit', data)
-  }
 
-  const handleCancel = () => {
-    return navigate(GAME_ROUTE)
-  }
+  const handleSubmit = async (userData: UserProfileInfo) => {
+    try {
+      const res = await changeProfile(userData);
+      if ('data' in res) {
```

Comment:
а так тут не сработает? Выглядит проще
```suggestion
      if (res.data) {
```

Reply:
У меня был тот же момент, res.data был с ошибкой, она уходит если добавить типизацию
const res: any = await changeProfile(userData);

Альтернативу any правда, не смогла подобрать, не уверена что в итоге лучше

Reply:
Согласен с @anastasia-shmeleva 

Этот результат из хука, него тип:

![Снимок экрана 2023-03-14 в 15 27 54](https://user-images.githubusercontent.com/64530309/225008603-ebe8f58f-97cb-4f4f-879d-27f4d6443fd5.png)

К тому же проверка res.data не гарантирует, что это поле есть:

```
const res = { data: 0 }
if (res.data) console.log('no print')
if ('data' in res) console.log('print') 
```

Reply:
Ок, если вам тут нужна проверка именно на то, что поле есть, а не на то, что в нём есть какой-то контент, то такая проверка вполне оправдана